src/libraries/System.Private.Xml.Linq/tests/misc/LoadSaveAsyncTests.cs

```diff
@@ -218,53 +218,184 @@ public async Task SaveAsync_CallsAsyncOnly_SaveSync_CallsSyncOnly(bool isAsync,
         {
             XDocument document = XDocument.Parse("<root>Test document async save</root>");
             var element = new XElement("Test");
-            using (ForceSyncAsyncStream stream = new ForceSyncAsyncStream(isAsync))
+            
+            if (isAsync)
             {
-                if (isAsync)
+                await using (ForceSyncAsyncStream stream = new ForceSyncAsyncStream(isAsync))
                 {
                     await document.SaveAsync(stream, saveOptions, CancellationToken.None);
                     await element.SaveAsync(stream, saveOptions, CancellationToken.None);
                 }
-                else
+            }
+            else
+            {
+                using (ForceSyncAsyncStream stream = new ForceSyncAsyncStream(isAsync))
                 {
                     document.Save(stream);
                     element.Save(stream);
                 }
             }
         }
+
+        [Theory]
+        [MemberData(nameof(RoundtripSyncAsyncOnly_MemberData))]
+        public static async Task RoundtripSyncAsyncOnly(bool async, bool document, LoadOptions loadOptions, SaveOptions saveOptions)
+        {
+            ForceSyncAsyncStream outputStream = new ForceSyncAsyncStream(async);
+            
+            if (async)
+            {
+                await using (Stream inputStream = new ForceSyncAsyncStream(async, FilePathUtil.getStream(GetTestFileName())))
+                {
+                    if (document)
+                    {
+                        XDocument asyncDoc = await XDocument.LoadAsync(inputStream, loadOptions, CancellationToken.None);
```

Comment:
Could we have test for XmlReader overload?

Reply:
Done. I also added a test for the StreamReader overload for completeness.

Reply:
Great